urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/driver/Phase1.java

```diff
@@ -0,0 +1,244 @@
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("CORRECT: " + ex.getMessage() + ": " + testPriceHolder.get(8) + " + " + testPriceHolder.get(0));
+        }
+        try {
+            testPriceHolder.get(8).subtract(testPriceHolder.get(0));
+            System.out.println("ERROR: Subtracting a LIMIT price from a MARKET Price succeeded: " + testPriceHolder.get(8) + " - " + testPriceHolder.get(0));
+
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("CORRECT: " + ex.getMessage() + ": " + testPriceHolder.get(8) + " - " + testPriceHolder.get(0));
+        }
+        try {
+            testPriceHolder.get(8).multiply(10);
+            System.out.println("ERROR: Multiplying a MARKET price succeeded: " + testPriceHolder.get(8) + " + 10");
+
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("CORRECT: " + ex.getMessage() + ": " + testPriceHolder.get(8) + " * 10");
+        }
+        System.out.println();
+    }
+
+    private static void verifyBooleanChecks() {
+        System.out.println("Verifying the Functionality of your Boolean Checks:");
+        System.out.println("Value      | Is Negative | Is Market");
+        System.out.println("------------------------------------");
+        String format = "%-9s  | %-12s| %-12s%n";
+        System.out.format(format, testPriceHolder.get(0), testPriceHolder.get(0).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(0).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(1), testPriceHolder.get(1).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(1).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(2), testPriceHolder.get(2).isNegative() ? "CORRECT" : "ERROR", testPriceHolder.get(2).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(3), testPriceHolder.get(3).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(3).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(4), testPriceHolder.get(4).isNegative() ? "CORRECT" : "ERROR", testPriceHolder.get(4).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(5), testPriceHolder.get(5).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(5).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(6), testPriceHolder.get(6).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(6).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(7), testPriceHolder.get(7).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(7).isMarket() ? "ERROR" : "CORRECT");
+        System.out.format(format, testPriceHolder.get(8), testPriceHolder.get(8).isNegative() ? "ERROR" : "CORRECT", testPriceHolder.get(8).isMarket() ? "CORRECT" : "ERROR");
+        System.out.println();
+    }
+
+    private static void verifyComparisons() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 37. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 137-168 into a separate method.